### ISSUE TITLE
Custom Template - KVM - qemu-system-x86_64

### ISSUE THREAD

**Issue by rahulmr** [kimchi-project/kimchi#639]
First of all, would like to thanks you people for creating such a wonderful project. :+1: 

Sorry, but I was not able to find any documentation on how to add or edit template which can create a custom VM. Below are the parameters which I want to add to create a Mac OS X Yosemite image using Kimchi.
```shell
qemu-system-x86_64 -enable-kvm -m 8192 -cpu core2duo \
      -machine q35 -smp 4,cores=2 \
      -usb -device usb-kbd -device usb-mouse -vga std \
      -device isa-applesmc,osk="ourhardworkbythesewordsguardedpleasedontsteal(c)AppleComputerInc" \
      -smbios type=2 \
      -vnc 0.0.0.0:1 \
      -device ide-drive,bus=ide.2,drive=MacHDD \
      -drive id=MacHDD,if=none,file=./mac_hdd_10.10niresh.img \
      -netdev user,id=hub0port0 \
      -device e1000-82545em,netdev=hub0port0,id=mac_vnet0
```

I want to create something like this. When I try to create a template it is asking for the .img file and then it is throwing error that the OS can not be detected and further fails to create the template.

Is there any way to patch the installation so that I can add my own parameters other than vnc?

Thanks for helping me out!

**Comment by cd1** [comment]
Hi Rahul!

I guess I didn't understand exactly what you're trying to do. Do you want to create a template for a MacOS X guest using an existing disk which you already have? If so, you can go to the tab "Templates", click on the button "+" in the top right corner, select "Local Image File", enter the full path of the image you have (i.e. `mac_hdd_10.10niresh.img`) and click on "Create". The template should be created in a few seconds. I've never tried doing this using a MacOS X disk image but it should work.

However, there's no way for you to provide specific QEMU parameters to run your VM, if that's what you want. Kimchi detects the OS used by the guest and provides different QEMU parameters for each one of them automatically. The only customizable items are the ones available when you edit a template (i.e. select an existing template, click on "Actions" > "Edit"). We are constantly adding more items there. There are many more items other than VNC which you can choose in that screen. But not everything can be customized.

Let us know if you were able to do what you want.

**Comment by rahulmr** [comment]
Yup, got it. The answer is No :) . You are right. I tried creating template but it is not recognized. I wanted to add qemu params, since I am able to work fine with qemu-system-x86_64 command. I just wanted to check whether I can access the VM using Kimchi Interface. Now I will try to create the vm using virt-manager and then see if it can be accessed from Kimchi interface. Thanks for your feedback. 
Further, I would like to say that it could be a good thing if it is possible to add a section where in some extra options can be provided for the VM. Like audio devices, etc. Proxmox VE we can add extra parameters.

**Comment by cd1** [comment]
Any VM created using libvirt (e.g. virt-manager, virsh) will be displayed and usable by Kimchi, so you can try that as well. However, we'll investigate on why creating a template with an existing MacOS X disk doesn't work.

We don't have plans to allow the user to add any custom parameter to the VMs, but if we change our minds, we'll let you know.

Thanks for your feedback!

**Comment by fahadysf** [comment]
Amen to this feature. The easiest starting point for this would be an "Edit Domain XML" feature for the existing VMs to start with, which pops up an in browser editor to edit the XML file and behaves like virsh edit <domain-name>.

At a later stage more customization options could be added into the template page itself. 

My use case was that i wanted to create Q35(ICH9) VMs by default instead of the i440FX + PIIX ones which current Kimchi templates create.
